explorer/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"encoding/gob"
+	"os"
+
+	hplugin "github.com/hashicorp/go-plugin"
+	"github.com/ignite/cli/ignite/services/plugin"
+
+	"explorer/cmd"
+)
+
+func init() {
+	gob.Register(plugin.Manifest{})
+	gob.Register(plugin.ExecutedCommand{})
+	gob.Register(plugin.ExecutedHook{})
+}
+
+type p struct{}
+
+func (p) Manifest() (plugin.Manifest, error) {
+	m := plugin.Manifest{
+		Name: "explorer",
+	}
+	m.ImportCobraCommand(cmd.NewExplorer(), "ignite")
+	return m, nil
+}
+
+func (p) Execute(c plugin.ExecutedCommand) error {
+	// Instead of a switch on c.Use, we run the root command like if
+	// we were in a command line context. This implies to set os.Args
+	// correctly.
+	// Remove the first arg "ignite" from OSArgs because our explorer
+	// command root is "explorer" not "ignite".
+	os.Args = c.OSArgs[1:]
+	return cmd.NewExplorer().Execute()
+}
+
+func (p) ExecuteHookPre(_ plugin.ExecutedHook) error {
+	return nil
+}
+
+func (p) ExecuteHookPost(_ plugin.ExecutedHook) error {
+	return nil
+}
+
+func (p) ExecuteHookCleanUp(_ plugin.ExecutedHook) error {
+	return nil
+}
```

Comment:
Maybe:

```suggestion
func (p) ExecuteHookPre(plugin.ExecutedHook) error {
	return nil
}

func (p) ExecuteHookPost(plugin.ExecutedHook) error {
	return nil
}

func (p) ExecuteHookCleanUp(plugin.ExecutedHook) error {
	return nil
}
```

But if you want to remove them completely I think you could do so if you define the plugin as:

```go
type p struct{
	plugins.Interface
}
```

Reply:
I don't think this gonna work since by doing this we don't actually implement those methods, right? so if the plugin library call one of those functions it will panic.

Reply:
both options have the same result

Reply:
> I don't think this gonna work since by doing this we don't actually implement those methods, right? so if the plugin library call one of those functions it will panic.

Yes, I think you are right, but those methods won't be called as long as you don't define any hooks in the manifest call.

Reply:
IMO, this doesn't sound very safe ( i mean the way we not define some of the behaviors of the code and then try to restrict using them in a configuration. Also embedding the `plugins.Interface` (an interface) to a struct and not using it to embed a variable is something you don't see much in go community.